lib/datastore/query.js

```diff
@@ -316,35 +287,27 @@ Query.prototype.offset = function(n) {
  *     stream instance is returned.
  * @param {?error} callback.err - An error returned while making this request
  * @param {object[]} callback.entities - A list of entities.
- * @param {?object} callback.nextQuery - If present, query with this object to
- *     check for more results.
- * @param {object} callback.apiResponse - The full API response.
+ * @param {object} callback.info - An object useful for pagination.
+ * @param {?string} callback.info.endCursor - Use this in a follow-up query to
+ *     begin from where these results ended.
+ * @param {string} callback.info.moreResults - Datastore responds with one of:
+ *     - `MORE_RESULTS_AFTER_LIMIT`: There *may* be more results after the
+ *       specified limit.
+ *     - `MORE_RESULTS_AFTER_CURSOR`: There *may* be more results after the
+ *       specified end cursor.
+ *     - `NO_MORE_RESULTS`: There are no more results.
  *
  * @example
- * query.run(function(err, entities) {});
- *
- * //-
- * // To control how many API requests are made and page through the results
- * // manually, call `autoPaginate(false)` on your query.
- * //-
- * query.autoPaginate(false);
- *
- * function callback(err, entities, nextQuery, apiResponse) {
- *   if (nextQuery) {
- *     // More results might exist.
- *     nextQuery.run(callback);
- *   }
- * }
- *
- * query.run(callback);
+ * query.run(function(err, entities, info) {});
  *
  * //-
- * // If you omit the callback, `run` will automatically call subsequent queries
- * // until no results remain. Entity objects will be pushed as they are found.
+ * // If you omit the callback, you will get the matching entities in a readable
+ * // object stream.
  * //-
  * query.run()
  *   .on('error', console.error)
  *   .on('data', function (entity) {})
+ *   .on('info', function(info) {})
```

Comment:
Any reason we don't just emit with this with `end`? Or is it assumed that if `info` is emitted the query is done running?